[v2, windows] Random panics during processRequest

**Description**
If there's a lot of WebResource processing ongoing some random panics occur. It seems like some COM syscalls return a nil pointer and don't set the pointer to the pointer onto the Go stack.

```go
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x607a94]

goroutine 1 [running, locked to thread]:
github.com/leaanthony/go-webview2/pkg/edge.(*ICoreWebView2WebResourceRequest).Release(0x0)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceRequest.go:45 +0x14
panic({0x895200, 0xd37b70})
	/opt/homebrew/Cellar/go/1.17.6/libexec/src/runtime/panic.go:1038 +0x215
github.com/leaanthony/go-webview2/pkg/edge.(*ICoreWebView2WebResourceRequest).GetUri(0x7ff8b67a8970)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceRequest.go:32 +0x3c
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.(*Frontend).processRequest(0xc00012e3f0, 0xc00039d3e8, 0x7b7a000a1560)
	/builds/app/vendor/github.com/wailsapp/wails/v2/internal/frontend/desktop/windows/frontend.go:349 +0x55
github.com/leaanthony/go-webview2/pkg/edge.(*Chromium).WebResourceRequested(0xc000388000, 0xc00039d530, 0xc00039d4a8)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/chromium.go:260 +0xbd
github.com/leaanthony/go-webview2/pkg/edge._ICoreWebView2WebResourceRequestedEventHandlerInvoke(0xc00039d4b8, 0x462c9d, 0x0)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceRequestedEventHandler.go:26 +0x22
syscall.Syscall(0x7ff911e31a60, 0x1, 0x15c1810, 0x0, 0x0)
	/opt/homebrew/Cellar/go/1.17.6/libexec/src/runtime/syscall_windows.go:483 +0xf4
syscall.(*Proc).Call(0x40d767, {0xc0003754b0, 0x873de0, 0x2bdd9e01})
	/opt/homebrew/Cellar/go/1.17.6/libexec/src/syscall/dll_windows.go:183 +0xc6
syscall.(*LazyProc).Call(0xc000155650, {0xc0003754b0, 0x1, 0x1})
	/opt/homebrew/Cellar/go/1.17.6/libexec/src/syscall/dll_windows.go:340 +0x50
github.com/leaanthony/winc/w32.DispatchMessage(...)
	/builds/app/vendor/github.com/leaanthony/winc/w32/user32.go:327
github.com/leaanthony/winc.RunMainLoop()
...
```

```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x6041b4]

goroutine 1 [running, locked to thread]:
github.com/leaanthony/go-webview2/pkg/edge.(*ICoreWebView2WebResourceResponse).Release(0x0)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceResponse.go:25 +0x14
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.(*Frontend).processRequest(0xc0001682d0, 0xc000107390, 0x1b06000b0780)
	/builds/app/vendor/github.com/wailsapp/wails/v2/internal/frontend/desktop/windows/frontend.go:400+0x519
github.com/leaanthony/go-webview2/pkg/edge.(*Chromium).WebResourceRequested(0xc00006af00, 0xc0001074d8, 0xc000107450)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/chromium.go:260 +0xbd
github.com/leaanthony/go-webview2/pkg/edge._ICoreWebView2WebResourceRequestedEventHandlerInvoke(0xc000107460, 0x46299d, 0x0)
	/builds/app/vendor/github.com/leaanthony/go-webview2/pkg/edge/ICoreWebView2WebResourceRequestedEventHandler.go:26 +0x22
syscall.Syscall(0x7ff954bc1a60, 0x1, 0x12924e0, 0x0, 0x0)
	/opt/homebrew/Cellar/go/1.17.5/libexec/src/runtime/syscall_windows.go:483 +0xf4
syscall.(*Proc).Call(0x40d507, {0xc0003907c0, 0x86b1a0, 0x2be64601})
	/opt/homebrew/Cellar/go/1.17.5/libexec/src/syscall/dll_windows.go:183 +0xc6
syscall.(*LazyProc).Call(0xc00017f680, {0xc0003907c0, 0x1, 0x1})
	/opt/homebrew/Cellar/go/1.17.5/libexec/src/syscall/dll_windows.go:340 +0x50
github.com/leaanthony/winc/w32.DispatchMessage(...)
	/builds/app/vendor/github.com/leaanthony/winc/w32/user32.go:327
github.com/leaanthony/winc.RunMainLoop()
....
```

**To Reproduce**
Steps to reproduce the behaviour: Create a demo app with several resources and continuously reload the window, e.g. every second.

**Additional context**
After some investigation I've found out that `go-webview2` doesn't satisfy all Go [unsafe.Pointer](https://pkg.go.dev/unsafe#Pointer) rules. I've provided PR leaanthony/go-webview2#7 to fix those issues.

--- a/v2/go.mod
+++ b/v2/go.mod
@@ -22,7 +22,7 @@ require (
 	github.com/leaanthony/debme v1.2.1
 	github.com/leaanthony/go-ansi-parser v1.0.1
 	github.com/leaanthony/go-common-file-dialog v1.0.3
-	github.com/leaanthony/go-webview2 v1.0.2
+	github.com/leaanthony/go-webview2 v1.0.3-0.20220125181555-00f20a61c17d
 	github.com/leaanthony/gosod v1.0.3
 	github.com/leaanthony/idgen v1.0.0
 	github.com/leaanthony/slicer v1.5.0

--- a/v2/internal/frontend/desktop/windows/frontend.go
+++ b/v2/internal/frontend/desktop/windows/frontend.go
@@ -280,6 +280,9 @@ func (f *Frontend) Quit() {
 func (f *Frontend) setupChromium() {
 	chromium := edge.NewChromium()
 	f.chromium = chromium
+	if opts := f.frontendOptions.Windows; opts != nil && opts.WebviewUserDataPath != "" {
+		chromium.DataPath = opts.WebviewUserDataPath
+	}
 	chromium.MessageCallback = f.processMessage
 	chromium.WebResourceRequestedCallback = f.processRequest
 	chromium.NavigationCompletedCallback = f.navigationCompleted

--- a/v2/pkg/options/windows/windows.go
+++ b/v2/pkg/options/windows/windows.go
@@ -8,4 +8,8 @@ type Options struct {
 
 	// Draw a border around the window, even if the window is frameless
 	EnableFramelessBorder bool
+
+	// Path where the WebView2 stores the user data. If empty %APPDATA%\[BinaryName.exe] will be used.
+	// If the path is not valid, a messagebox will be displayed with the error and the app will exit with error code.
+	WebviewUserDataPath string
 }

--- a/website/docs/reference/options.mdx
+++ b/website/docs/reference/options.mdx
@@ -46,6 +46,7 @@ func main() {
             WindowIsTranslucent:   false,
             DisableWindowIcon:     false,
             EnableFramelessBorder: false,
+            WebviewUserDataPath:   "",
         },
         Mac: &mac.Options{
             TitleBar: &mac.TitleBar{
@@ -362,6 +363,14 @@ Type: bool
 
 Setting this to `true` will add a border around the window if [Frameless](#Frameless) has been activated.
 This allows hiding the title bar but still having a border around the window.
+
+### WebviewUserDataPath
+
+Name: WebviewUserDataPath
+
+Type: string
+
+This defines the path where the WebView2 stores the user data. If empty `%APPDATA%\[BinaryName.exe]` will be used.
 
 ## Mac Specific Options
 

